packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/cluster_client.ts

```diff
@@ -47,25 +48,33 @@ export class ClusterClient implements ICustomClusterClient {
     authHeaders,
     getExecutionContext = noop,
     getUnauthorizedErrorHandler = noop,
+    agentManager,
   }: {
     config: ElasticsearchClientConfig;
     logger: Logger;
     type: string;
     authHeaders?: IAuthHeadersStorage;
     getExecutionContext?: () => string | undefined;
     getUnauthorizedErrorHandler?: () => UnauthorizedErrorHandler | undefined;
+    agentManager?: AgentManager;
```

Comment:
Having it optional is an easy way to fallback to the previous behavior (as per my comment above).

Reply:
when an agentManager is not supplied I'd say our clients no longer behave as expected e.g. the meaning of maxSockets will change. It's only core using this constructor so it would be easy to change if we switch to undici.

Reply:
Fair enough, I'll make it mandatory.